pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -281,10 +281,14 @@ func (b *AutoscalingGroupModelBuilder) buildLaunchTemplateTask(c *fi.CloudupMode
 
 	if ig.Spec.InstanceMetadata != nil && ig.Spec.InstanceMetadata.HTTPPutResponseHopLimit != nil {
 		lt.HTTPPutResponseHopLimit = ig.Spec.InstanceMetadata.HTTPPutResponseHopLimit
+	} else if ig.IsControlPlane() && (b.Cluster.IsKubernetesLT("1.27") || !b.UseServiceAccountExternalPermissions()) {
+		lt.HTTPPutResponseHopLimit = fi.PtrTo[int64](3)
```

Comment:
Previously, clusters created long ago and successively upgraded to k8s 1.26 would have the hop limit defaulting to 1, but with HTTP tokens optional. With this change, non-IRSA control-plane nodes in such clusters will have the hop limit set to 3 (but HTTP tokens still optional).

This is technically a change in behavior for k8s < 1.27, though it might not matter.